### PR TITLE
[AKS] `az aks create/update`: Add `--nat-gateway-managed-outbound-ip-count` and `--nat-gateway-idle-timeout` to support nat gateway integration

### PR DESCRIPTION
**Pull request by MartinForReal**
Signed-off-by: MartinForReal <fanshangxiang@gmail.com>

**Description**
Add nat gateway integration for aks service

**Testing Guide**

```
az aks create -g MyResourceGroup -n MyManagedCluster --nat-gateway-managed-outbound-ip-count 1 --nat-gateway-idle-timeout 4 --outbound-type managedNATGateway
```

```
az aks update -g MyResourceGroup -n MyManagedCluster --nat-gateway-managed-outbound-ip-count 2 --nat-gateway-idle-timeout 10`
```

**History Notes**

[AKS] `az aks create`: Add `--nat-gateway-managed-outbound-ip-count` and `--nat-gateway-idle-timeout` and Add `managedNATGateway` and `userAssignedNATGateway` to supported outbound type

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
acs

**Review by FumingZhang — APPROVED** [review]
LGTM

**Comment by wangzelin007** [comment]
@MartinForReal,
Pelease fix style error to pass CI checks.
image

**Comment by MartinForReal** [comment]
Sure I will.

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -489,6 +489,37 @@ def test_aks_create_default_service_with_skip_role_assignment(self, resource_gro
...
+            'name': aks_name,
+            'ssh_key_value': self.generate_ssh_keys(),
+            'location': resource_group_location
+        })
+
+        create_cmd = 'aks create --resource-group={resource_group} --name={name} ' \
+                     '--vm-set-type VirtualMachineScaleSets -c 1 ' \
+                     '--outbound-type=managedNATGateway ' \
+                     '--generate-ssh-keys ' \
+                     '--location={location}'
```
Comment (wangzelin007) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py:507]: Please also test `--nat-gateway-managed-outbound-ip-count` and `--nat-gateway-idle-timeout` in `az aks create` command.

**Review by wangzelin007 — APPROVED** [review]

**Review by andyliuliming — APPROVED** [review]

File: src/azure-cli-core/azure/cli/core/_session.py [hunk 2]
```diff
@@ -7,12 +7,19 @@
...
+
+try:
+    collectionsAbc = collections.abc
+except AttributeError:
+    collectionsAbc = collections
+
+
+class Session(collectionsAbc.MutableMapping):
 
-class Session(MutableMapping):
```
Comment (zhoxing-ms) [inline src/azure-cli-core/azure/cli/core/_session.py:15]: I believe this issue has been solved by PR #21877. @jiasli Could you help confirm it?

Comment (MartinForReal) [reply src/azure-cli-core/azure/cli/core/_session.py]: I've rebased my branch onto the dev branch. Thanks for the info!

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**